presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -163,6 +180,14 @@ public DispatchManager(
     public void start()
     {
         queryTracker.start();
+        queryHeartbeatExecutor.scheduleWithFixedDelay(() -> {
+            try {
+                sendQueryHeartbeats();
+            }
+            catch (Throwable e) {
+                log.error(e, "Error sending Heartbeat Event");
```

Comment:
Issues in sending the heartbeat should be caught and handled by QueryMonitor, DispatchManager shouldn't worry about that or log anything.